FLT/Deformations/Algebra/InverseLimit/Basic.lean

```diff
@@ -0,0 +1,472 @@
+/-
+Copyright (c) 2025 Javier López-Contreras. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Javier López-Contreras
+-/
+import Mathlib.Algebra.Module.LinearMap.Defs
+import Mathlib.Data.Rat.Cast.Defs
+import Mathlib.Order.DirectedInverseSystem
+import Mathlib.Tactic.SuppressCompilation
+
+/-!
+# Inverse limit of algebraic structures
+
+We introduce all kinds of algebraic instances on `InverseLimit`, and specialize to the cases
+of modules and rings, showing that they are indeed limits in the respective categories.
+-/
+
+suppress_compilation
+
+variable {R ι : Type*} [Preorder ι] {G : ι → Type*}
+variable {T : ∀ ⦃i j : ι⦄, i ≤ j → Type*} {f : ∀ _ _ h, T h}
+variable [∀ i j (h : i ≤ j), FunLike (T h) (G j) (G i)] [InverseSystem (f · · ·)]
+variable [IsDirected ι (· ≤ ·)]
```

Comment:
```suggestion
variable [∀ i j (h : i ≤ j), FunLike (T h) (G j) (G i)]
```
You never use these hypotheses. You put them at the beginning of the file and then spend literally the entire rest of the file not using them.